config/src/keymap/key.rs

```diff
@@ -32,13 +32,28 @@ impl Default for Key {
 
 impl From<KeyEvent> for Key {
 	fn from(value: KeyEvent) -> Self {
-		let shift = matches!(value.code, KeyCode::Char(c) if c.is_ascii_uppercase());
+		// For alphabet:
+		//   Unix    :  <S-a> => Char("A") + SHIFT
+		//   Windows :  <S-a> => Char("A") + SHIFT
+		//
+		// For non-alphabet:
+		//   Unix    :  <S-`> => Char("~") + NULL
+		//   Windows :  <S-`> => Char("~") + SHIFT
+		//
+		// So we detect `Char("~") + SHIFT`, and change it to `Char("~") + NULL`
+		// for consistent behavior between OSs.
+
+		let shift = match (value.code, value.modifiers) {
+			(KeyCode::Char(c), _) if c.is_ascii_uppercase() => true,
+			(KeyCode::Char(_), m) if m.contains(KeyModifiers::SHIFT) => false,
```

Comment:
Seems these two lines can be merged into `(KeyCode::Char(c), _) => c.is_ascii_uppercase()`?

Reply:
Yes, that's right

Reply:
Done